File: back/src/main/java/com/back/domain/ticket/dto/PaymentTicketResponse.java
package com.back.domain.ticket.dto;

import com.back.domain.schedule.entity.Schedule;
import com.back.domain.schedule.entity.ScheduleSeat;
import com.back.domain.schedule.entity.SeatStatus;
import com.back.domain.ticket.entity.Ticket;

import java.time.LocalDateTime;

public record PaymentTicketResponse(
        String ticketNumber,
        String urlPoster,
        String concertName,
        String seatNumber,
        LocalDateTime scheduleDate,
        SeatStatus seatStatus,
        boolean isValid
) {
    public static PaymentTicketResponse from(ScheduleSeat scheduleSeat, Schedule schedule, Ticket ticket) {
        return new PaymentTicketResponse(
                ticket.getTicketNumber(),
                schedule.getConcert().getUrlPoster(),
                schedule.getConcert().getConcertName(),
                scheduleSeat.getSeatNumber(),
                schedule.getScheduleDate(),
                scheduleSeat.getSeatStatus(),
                ticket.isValid()
        );
    }
}

File: back/src/main/kotlin/com/back/domain/ticket/dto/PaymentTicketResponse.kt
package com.back.domain.ticket.dto

import com.back.domain.schedule.entity.Schedule
import com.back.domain.schedule.entity.ScheduleSeat
import com.back.domain.schedule.constant.SeatStatus
import com.back.domain.ticket.entity.Ticket
import java.time.LocalDateTime

data class PaymentTicketResponse(
    val ticketNumber: String,
    val urlPoster: String?,
    val concertName: String,
    val seatNumber: String,
    val scheduleDate: LocalDateTime,
    val seatStatus: SeatStatus,
    val isValid: Boolean
) {
    companion object {
        fun from(scheduleSeat: ScheduleSeat, schedule: Schedule, ticket: Ticket): PaymentTicketResponse {
            return PaymentTicketResponse(
                ticketNumber = ticket.ticketNumber,
                urlPoster = schedule.concert.urlPoster,
                concertName = schedule.concert.concertName,
                seatNumber = scheduleSeat.seatNumber,
                scheduleDate = schedule.scheduleDate,
                seatStatus = scheduleSeat.seatStatus,
                isValid = ticket.isValid
            )
        }
    }
}
